src/digital.rs

```diff
@@ -2,19 +2,45 @@
 
 /// Single digital output pin
 pub trait OutputPin {
-    /// Is the output pin high?
-    fn is_high(&self) -> bool;
-
-    /// Is the output pin low?
-    fn is_low(&self) -> bool;
-
     /// Sets the pin low
     fn set_low(&mut self);
 
     /// Sets the pin high
     fn set_high(&mut self);
 }
 
+/// Output pin that can read its output state
+#[cfg(feature = "unproven")]
+trait StatefulOutputPin {
+    /// Is the pin set to high?
+    fn is_set_high(&self) -> bool;
+
+    /// Is the pin set to low?
+    fn is_set_low(&self) -> bool;
+}
+
+/// Output pin that can be toggled
+#[cfg(feature = "unproven")]
+trait ToggleableOutputPin {
+    /// Toggle pin output
+    fn toggle(&mut self);
+}
+
+/// If you can read **and** write the output state, a pin is
+/// toggleable by software. You may override the `toggle()` method
+/// with a hardware implementation.
+#[cfg(feature = "unproven")]
+impl<PIN: OutputPin + StatefulOutputPin> ToggleableOutputPin for PIN {
```

Comment:
Would it be good to add an empty `DefaultToggle` trait here, and thus make this implementation opt-in rather than opt-out?

It would match the way the blocking SPI traits are handled. On the other hand, this is a simpler implementation.

I'm not particularly attached to one way or the other, but I do think consistency throughout the `embedded-hal` will be valuable.

Reply:
Thanks for the suggestion. I added that now by copying the style of the opt-in default traits in `spi` and `serial`. I hope that the doc example is helpful.